src/elmAnalyse.ts

```diff
@@ -0,0 +1,240 @@
+import {isWindows, execCmd, ExecutingCmd } from './elmUtils';
+import {runLinter, IElmIssue} from './elmLinter';
+
+export class ElmAnalyse {
+  public constructor(public elmAnalyseIssues: IElmIssue[]) {
+    this.statusBarStopButton = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left);
+    this.statusBarInformation = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left);
+    this.statusBarStopButton.text = '$(primitive-square)' + ' Stop Elm-analyse process';
+    this.statusBarStopButton.command = 'elm.analyseStop';
+    this.statusBarStopButton.tooltip = 'Stop elm-analyse process';
+    this.analyse = {} as ExecutingCmd;
+    this.messageDescriptionsMap = new Map()
+    this.messageDescriptions.forEach(element => {
+      this.messageDescriptionsMap.set(element.messageType, element.description)
+    });
+    const config: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('elm');
+    const enabledOnStartup: boolean = <boolean>config.get('analyseEnabled');
+    if (enabledOnStartup) {
+      this.activateAnalyseProcesses();
+    }
+   }
+  private statusBarStopButton: vscode.StatusBarItem;
+  private statusBarInformation: vscode.StatusBarItem;
+  private analyseSocket: WebSocket;
+  private analyse: ExecutingCmd;
+  private oc: vscode.OutputChannel = vscode.window.createOutputChannel('Elm Analyse');
+  private messageDescriptionsMap : Map<string, string>
+
+  private initSocketClient() {
+    try {
+      const cwd: string = vscode.workspace.rootPath;
+      const config: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('elm');
+      const analyseCommand: string = <string>config.get('analyseCommand');
+      const port: string = <string>config.get('analysePort');
+      const wsPath = "ws://localhost:" + port + "/state";
+      if (this.analyseSocket) {
+        this.analyseSocket.close();
+      }
+      this.analyseSocket = new WebSocket(wsPath);
+      this.analyseSocket.on("open", () => {
+        this.analyseSocket.send("PING");
+        this.statusBarInformation.text = 'elm-analyse websocket listening on port ' + port;
+        this.statusBarInformation.show();
+      });
+      this.analyseSocket.on("message", (stateJson) => {
+        try {
+          this.elmAnalyseIssues = [];
+          const state = JSON.parse(stateJson);
+          const messages = state.messages;
+          messages.forEach(message => {
+            if (message.hasOwnProperty("data") && message.data.hasOwnProperty("type")) {
+              let messageType = message.data.type;
+              if (message.data.hasOwnProperty(messageType)) {
+                let messageTypeInfo = message.data[messageType];
+                let messageInfoFileSrc = messageTypeInfo.file;
+                let messageInfoFileRange = [0, 0, 0, 0];
+                if (messageTypeInfo.hasOwnProperty("range")) {
+                  messageInfoFileRange = messageTypeInfo.range.real;
+                }
```

Comment:
It maybe better to extract this logic into a function. That would be the right place to implement the cases where the range is not set on the "range" property of the message. Working on the API docs at the moment, this will become more clear soon.